From 64fcc6fff124dd74e059b9271ba6693a2d561704 Mon Sep 17 00:00:00 2001
From: Joshua Goins <josh@redstrate.com>
Date: Tue, 21 May 2024 17:15:14 -0400
Subject: [PATCH] KMessageWidget: Avoid calling polish() inside of palette
 update

This causes infinite recursion as the palette is set and unset when the
application uses CSS stylesheets. But this call is simply unnecessary
and can be removed.

BUG: 487247
---
 src/kmessagewidget.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kmessagewidget.cpp b/src/kmessagewidget.cpp
index 762507c0d..793c2bb00 100644
--- a/src/kmessagewidget.cpp
+++ b/src/kmessagewidget.cpp
@@ -184,7 +184,7 @@ void KMessageWidgetPrivate::setPalette()
     // palette propagation
     iconLabel->setPalette(palette);
     textLabel->setPalette(palette);
-    q->style()->polish(q);
+
     // update the Icon in case it is recolorable
     q->setIcon(icon);
     q->update();
-- 
GitLab

